skfda/misc/metrics/_weighted_lp_norm.py

```diff
@@ -0,0 +1,146 @@
+"""Implementation of Weighted Lp norms."""
+
+from collections.abc import Callable
+
+import numpy as np
+
+from ..._utils import nquad_vec
+from ...representation import FData, FDataBasis, FDataGrid
+from ...typing._base import (
+    GridPointsLike,
+)
+from ...typing._metric import Norm
+from ...typing._numpy import NDArrayFloat
+
+
+class WeightedLpNorm:
+    def __init__(
+        self,
+        p: float,
+        vector_norm: Norm[NDArrayFloat] | float | None = None,
+        lp_weight: (
+            Callable[[GridPointsLike], NDArrayFloat] | float | None
+        ) = None,
+    ) -> None:
+
+        # Checks that the lp normed is well defined
```

Comment:
I would love if the code of Lp and this one could be merged together or simplified in some way.